kratos/python_scripts/process_factory.py

```diff
@@ -18,24 +19,7 @@ def ConstructListOfProcesses( self, process_list ):
             if item.Has("name"):
                 registry_entry = item["name"].GetString()
                 if KM.Registry.HasItem(registry_entry):
-                    # Get already stored prototype
-                    if KM.Registry.HasItem(f"{registry_entry}.Prototype"):
-                        prototype = KM.Registry[f"{registry_entry}.Prototype"]
-                    # Get prototype from stored Python module
-                    elif KM.Registry.HasItem(f"{registry_entry}.ModuleName"):
-                        class_name = registry_entry.split(".")[-1]
-                        module_name = KM.Registry[f"{registry_entry}.ModuleName"]
-                        module = import_module(module_name)
-                        if hasattr(module, class_name):
-                            prototype = getattr(module, class_name)
-                        else:
-                            #TODO: In here we're assuming that the registry last key is the class name
-                            #TODO: We should enforce this. Now an error happens but as we populate we should throw a warning and search for a ClassName item
-                            err_msg = f"The '{class_name}' class name cannot be found within the '{module_name}' module."
-                            raise Exception(err_msg)
-                    else:
-                        err_msg = f"Registry process '{registry_entry}' cannot be constructed."
-                        raise Exception(err_msg)
+                    prototype = self._GetPrototype(registry_entry)
                     # Construct the process from the obtained prototype and append it to the list
                     instance = prototype(self.Model, item["Parameters"])
                     constructed_processes.append(instance)
```

Comment:
i would say there is a problem with this: 

a process may need a variable to be in the modelpart already at construction time (i am afraid there were some examples of this).

for this reason one cannot use a "real" constructor of the process to get a prototype on the top of which we could get the function, without having constructed the real process.
What i am trying to say that the GetVariable should be a function similar to the Create one

I am cc-ing  @KratosMultiphysics/technical-committee since this was a thing we discussed.

Reply:
Any advance on this, or do I need to find another solution on our side?